src/caret_analyze/common/type_check_decorator.py

```diff
@@ -0,0 +1,160 @@
+            expected_types_str = str(expected_types)
+        else:
+            expected_types_str = f"'{expected_types[0]}'"
+
+        return expected_types_str
+
+    def _get_given_arg_loc(given_arg_loc: tuple) -> str:
+        """
+        Get given argument location.
+
+        Parameters
+        ----------
+        given_arg_loc: tuple
+            (i) Not iterable type case
+                ('[ARGUMENT_NAME],')
+
+            (ii) Iterable type except for dict case
+                ('[ARGUMENT_NAME]', '[INDEX]')
+
+            (ii) Dict case
+                ('[ARGUMENT_NAME]', '[KEY]')
+
+        """
+        if len(given_arg_loc) == 2:  # Iterable type case
+            loc_str = f"'{given_arg_loc[0]}'[{given_arg_loc[1]}]"
+        else:
+            loc_str = f"'{given_arg_loc[0]}'"
+
+        return loc_str
+
+    def _get_given_arg_type(
+        signature: Signature,
+        args: Tuple[Any, ...],
+        kwargs: Dict[str, Any],
+        given_arg_loc: tuple
+    ) -> str:
+        """
+        Get given argument type.
+
+        Parameters
+        ----------
+        signature: Signature
+            Signature of target function.
+        args: Tuple[Any, ...]
+            Arguments of target function.
+        kwargs: Dict[str, Any]
+            Keyword arguments of target function.
+        given_arg_loc: tuple
+            (i) Not iterable type case
+                ('[ARGUMENT_NAME],')
+
+            (ii) Iterable type except for dict case
+                ('[ARGUMENT_NAME]', '[INDEX]')
+
+            (ii) Dict case
+                ('[ARGUMENT_NAME]', '[KEY]')
+
+        """
+        arg_name = given_arg_loc[0]
+        given_arg: Any = None
+
+        # Check kwargs
+        for k, v in kwargs.items():
+            if k == arg_name:
+                given_arg = v
+                break
+
+        if given_arg is None:
+            # Check args
+            given_arg_idx = list(signature.parameters.keys()).index(arg_name)
+            given_arg = args[given_arg_idx]
+
+        if len(given_arg_loc) == 2:  # Iterable type case
+            if isinstance(given_arg, dict):
+                given_arg_type_str = f"'{given_arg[given_arg_loc[1]].__class__.__name__}'"
+            else:
+                given_arg_type_str = f"'{given_arg[int(given_arg_loc[1])].__class__.__name__}'"
+        else:
+            given_arg_type_str = f"'{given_arg.__class__.__name__}'"
+
+        return given_arg_type_str
+
+    def decorator(func):
+        validate_arguments_wrapper = \
+            validate_arguments(config={'arbitrary_types_allowed': True})(func)
+
+        @wraps(func)
+        def _custom_wrapper(*args, **kwargs):
+            try:
+                return validate_arguments_wrapper(*args, **kwargs)
+            except ValidationError as e:
+                expected_types = _get_expected_types(e)
+                loc_tuple = e.errors()[0]['loc']
+                given_arg_loc = _get_given_arg_loc(loc_tuple)
```

Comment:
nits

It might `arg_name` is more readable.

```suggestion
                given_arg_name = _get_given_arg_name(loc_tuple)
```

Reply:
Thank you very much for pointing that out.
Since the location of the argument is formatted as a string, I thought that `arg_loc_str` would be more appropriate than `arg_name`.
Fixed in: https://github.com/tier4/CARET_analyze/pull/186/commits/ca2397ac7949114cd65e29f0d41713d1a5ed3e1e